lsp-iedit.el

```diff
@@ -0,0 +1,55 @@
+;;; lsp-iedit.el --- `iedit' integration -*- lexical-binding: t -*-
+
+;;; Commentary:
+
+;; This module provides features that allow starting `iedit' on various
+;; different lsp-based, semantic units (like documentHighlights, and
+;; linkedEditingRanges in the future).
+
+;;; Code:
+
+(require 'lsp-mode)
+
+(defvar iedit-mode)
+(defvar iedit-auto-buffering)
+(defvar iedit-occurrences-overlays)
+(declare-function iedit-make-occurrence-overlay "iedit-lib" (begin end))
+(declare-function iedit-start-buffering "iedit-lib" ())
+(defun lsp-iedit--on-ranges (ranges)
+  "Start an `iedit' operation using RANGES.
+RANGES shall be a list of lsp-`&Range's. They can be acquired
+from various lsp protocol requests, e.g.
+`textDocument/documentHighlight', ...."
+  (require 'iedit)
```

Comment:
Maybe its better to change to a:
```elisp
(if (require 'iedit nil t)
      ....
   (lsp-warn "iedit package is not installed"))
```

Reply:
This would have to be an `lsp-warn` followed by an `error` anyway, since we mustn't continue trying to use `iedit`-functions without it loaded. Besides, `lsp-avy-lens` doesn't do that either. And lastly, I'd argue that such a warning is redundant, given that users of that function would probably already know about `iedit` and have it installed.

Reply:
Yeah, a lsp-error would fit better. 
I think a nice and clear UX for users is always better :)

Reply:
I absolutely agree that we should strive for having a good and intuitive UX. However, I don't believe using `lsp--error` is the right thing here, because it conflicts with doing things in a way that is consistent with the rest of Emacs. Few (perhaps none?) packages are putting `require nil t` + `(warning)` guards around `require`, since that is cumbersome. Calling this function without `iedit` installed should be a regular `error`, not just demoted to a pretty `lsp` message. And the error message from an invalid `require` isn't too bad: `(file-missing "Cannot open load file" "No such file or directory" "q")`.

Reply:
I see, my point is user may not know what is `iedit` and wanna try that lsp feature and understand what it does.
And a `Package x is not installed` may be a better message than the error that may happen at any time.
But is just a suggestion, no hard feeling about that